lib/internal/main/worker_thread.js

```diff
@@ -111,6 +113,17 @@ port.on('message', (message) => {
     publicWorker.parentPort = publicPort;
     publicWorker.workerData = workerData;
 
+    // The counter is only passed to the workers created by the main thread, not
+    // to workers created by other workers.
+    let cachedCwd = '';
+    const originalCwd = process.cwd;
+
+    process.cwd = function() {
+      cachedCwd = originalCwd();
```

Comment:
I don't think we have a proper convention around this, but I'd add a comment explaining this is only for v12.x to reduce conflicts - this is where the difference between the two branches lie so it'll create conflicts when future commits touch here anyway.

Reply:
I struggle finding brief words describing this. I could just add `// Backwards compatibility patch: removed code to prevent breaking change.`.

I personally do not think it would be bad without the comment, since we just work on `master` and people backporting code should see the compatibility commit.